viadot/utils.py

```diff
@@ -83,29 +109,37 @@ def handle_api_response(
     params: Dict[str, Any] = None,
     headers: Dict[str, Any] = None,
     timeout: tuple = (3.05, 60 * 30),
+    method: Literal["GET", "POST", "DELETE"] = "GET",
+    data: str = None,
 ) -> requests.models.Response:
     """
     Handle an HTTP response by applying retries and handling some common response
     codes.
 
     Args:
-        url (str): the URL which trying to connect.
-        auth (tuple, optional): authorization information. Defaults to None.
-        params (Dict[str, Any], optional): the request parameters. Defaults to None.
-        headers: (Dict[str, Any], optional): the request headers. Defaults to None.
-        timeout (tuple, optional): the request times out. Defaults to (3.05, 60 * 30).
+        url (str): The URL which trying to connect.
+        auth (tuple, optional): Authorization information. Defaults to None.
+        params (Dict[str, Any], optional): The request parameters. Defaults to None.
+        headers: (Dict[str, Any], optional): The request headers. Defaults to None.
+        timeout (tuple, optional): The request times out. Defaults to (3.05, 60 * 30).
```

Comment:
typo: times out->timeout. Also the original description is better as it explains what each value is: `A tuple of (connect_timeout, read_timeout) in seconds.`